### PR TITLE
Storage get count

### PR DESCRIPTION
**Pull request by Simonster04**
Added get and count methods to file, db storage.
Start API and return status
Improve functions comments
Add Status of the API
Create relational .gitignore
Upgrade models relationship
Refine view User, Amenity, City, Place...
Place_review, State, User that handles all default RestFul API actions
Makes a cross-origin, the HTTP access control (CORS)

### CODE REVIEW

File: models/engine/file_storage.py [hunk 1]
```diff
@@ -68,3 +68,20 @@ def delete(self, obj=None):
...
         self.reload()
+
+    def get(self, cls, id):
+        """retrieve one object"""
+        if cls.__name__ in classes and id is not None:
+            objs = self.all(cls.__name__)
+            for key in objs.keys():
+                if key == cls.__name__ + '.' + id:
+                    return objs[key]
+        return None
```
Comment (02KellyV) [inline models/engine/file_storage.py:79]: You can optimize all this part +74 to +79 using the .`get(key)` dictionary function, where `key ` is class name + object id
```suggestion
        def get(self, cls, id):
            clss = self.all(cls)
            key = cls + '.' + id
            return clss.get(key)
```

Comment (02KellyV) [inline models/engine/file_storage.py]: You can optimize all this part +74 to +79 using the .`get(key)` dictionary function, where `key ` is `class name + object id`.

```suggestion
    def get(self, cls, id):
        """def that retrieve one object"""
        clss = self.all(cls)
        key = cls + '.' + id
        return clss.get(key)
```

File: models/engine/db_storage.py [hunk 2]
```diff
@@ -74,3 +74,22 @@ def reload(self):
...
         """call remove() method on the private session attribute"""
         self.__session.remove()
+
+    def get(self, cls, id):
+        """retrieve one object"""
+        if cls.__name__ in classes and id is not None:
+            return self.__session.query(classes[cls.__name__])\
+            .filter(classes[cls.__name__].id == id).first()
+        else:
+            return None
```
Comment (02KellyV) [inline models/engine/db_storage.py]: Protip: Same sug that in [file_storage comment](https://github.com/02KellyV/AirBnB_clone_v3/pull/1/files#r416317606).

File: models/engine/file_storage.py [hunk 3]
```diff
@@ -68,3 +68,20 @@ def delete(self, obj=None):
...
+
+    def count(self, cls=None):
+        """count the number of objects in storage"""
+        if cls is None:
+            return len(self.__objects)
+        if cls.__name__ in classes:
+            return len(self.all(classes[cls.__name__]))
+        return 0
```
Comment (02KellyV) [inline models/engine/file_storage.py]: You can do this code in max 3 lines using `len(obj)` function, where `obj` is our `clss`.

```suggestion
    def count(self, cls=None):
        """def that count the number of objects in storage"""
        return len(self.all(cls))
```

File: models/engine/db_storage.py [hunk 4]
```diff
@@ -74,3 +74,22 @@ def reload(self):
...
+    def count(self, cls=None):
+        """count the number of objects in storage"""
+        if cls is None:
+            num_objs = 0
+            for obj in classes.values():
+                num_objs += self.__session.query(obj).count()
+            return num_objs
+        if cls.__name__ in classes:
+            return self.__session.query(classes[cls.__name__]).count()
+        return 0
```
Comment (02KellyV) [inline models/engine/db_storage.py]: Protip: Same sug that in [file_storage comment](https://github.com/02KellyV/AirBnB_clone_v3/pull/1/files#r416323058).

File: models/engine/file_storage.py (hunk 1 above)
Comment (Simonster04) [reply models/engine/file_storage.py]: Excellent, I already update it with another change: I had to add the `__name__` special variable to `cls` as you can see below and now it works for me.

`key = cls.__name__ + '.' + id`

File: models/engine/file_storage.py (hunk 3 above)
Comment (Simonster04) [reply models/engine/file_storage.py]: Updated!

File: models/engine/file_storage.py (hunk 1 above)
Comment (02KellyV) [reply models/engine/file_storage.py]: I reviewed, but the cls is already an string. So,  `cls.__name__` is not necessary. 

**Error:**
```
File "/home/vagrant/AirBnB_clone_v3/models/engine/db_storage.py", line __, in count
    if cls.__name__ in classes:
AttributeError: 'str' object has no attribute '__name__'
```

**Review by 02KellyV — COMMENTED** [review]
All outputs works!

**Pull request merged**